make/gen.cmake

```diff
@@ -283,9 +298,9 @@ function(N_TEMPLATE)
   #
   # If APPLY_VERILOG_IMAGE_GEN, ADD_VERILOG_IMAGE_GEN will be invoked with each
   # output file.
-  set(options APPLY_V2X APPLY_VERILOG_IMAGE_GEN)
+  set(options APPLY_V2X APPLY_VERILOG_IMAGE_GEN DEPS)
   set(oneValueArgs NAME)
-  set(multiValueArgs SRCS PREFIXES)
+  set(multiValueArgs SRCS PREFIXES GENERATEDVERILOGDEPS GENERATEDXMLDEPS)
```

Comment:
Why should N_TEMPLATE need these arguments, it should all be computable from the SCANNER dependencies?

Reply:
the issue I had was that some xml files are included in other generated xml files, so SCANNER is able to determine the dependencies only for the first inclusion. The second (deeper) xml was newer added into dependencies.

Reply:
Correct, hence why append_file_dependencies is recursive.  Each file is scanned and points to the next up the chain.  ntemplate should recursively walk the INCLUDE_FILES lists and apply the ntemplate transformation, and then assign INCLUDE_FILES to the relevant targest.  V2X should be transforming the verilog INCLUDE_FILES list to XML  etc.

Reply:
hmm, the problem I had was that append_file_dependencies was not walking through the generated files included in other generated files (included in the first one)

Reply:
> hmm, the problem I had was that append_file_dependencies was not walking through the generated files included in other generated files (included in the first one)

This would happen if INCLUDE_FILES wasn't set on the generated file targets.  This is something that v2x and n_template need to handle.